lib/librato/metrics/aggregator.rb

```diff
@@ -52,20 +52,33 @@ def initialize(opts={})
       # @return [Aggregator] returns self
       def add(measurements)
         measurements.each do |metric, data|
+          entry = {}
           if @prefix
             metric = "#{@prefix}.#{metric}"
           end
+          entry[:name] = metric.to_s
           if data.respond_to?(:each) # hash form
+            validate_parameters(data)
```

Comment:
API should typically handle invalid params (e.g., attempting to submit a counter to `/v1/measurements`), but prefer to enforce at this level since it sets state (e.g., `@tags`, `@time`, `@source`, `@measure_time`).